rules/xcodeproj.bzl

```diff
@@ -151,12 +151,21 @@ def _xcodeproj_impl(ctx):
     xcodeproj_schemes_by_name = {}
     for target_info in targets:
         target_macho_type = "staticlib" if target_info.product_type == "framework" else "$(inherited)"
+        xcode_sources = []
+        for s in target_info.srcs.to_list():
+            src_info = {
+                            "path": paths.join(src_dot_dots, s.short_path),
+                            "group": paths.dirname(s.short_path),
+                            "optional": True,
+            }
+            if ".xcassets" in s.path:
```

Comment:
do we actually maybe want to filter out everything that would go into the `PBXResourcesBuildPhase`, instead of only asset catalogs?

Reply:
I wasn't sure how to figure out what all these possible things were, but I would have preferred to do that.

Reply:
Take a look at `_bucketize` in https://github.com/bazelbuild/rules_apple/blob/9c1e460ec08af6e36673d6c6f34c723b09ae9328/apple/internal/resources.bzl -- I think everything explicitly called out there (e.g. not `"unprocessed"`) could be ignore.

Another approach would be to propagate 2 different fields in the provider -- one for explicit `srcs` that come from the `srcs` attribute and another with basically everything else. Everything not in `srcs` would get added without a build phase. That's probably the more elegant way of handling this, rather than hardcoding based on extensions